src/delay.rs

```diff
@@ -0,0 +1,45 @@
+use hal::blocking::delay::{DelayMs, DelayUs};
+
+use lpc11uxx::SYST;
+use cortex_m::peripheral::syst::SystClkSource;
+
+pub struct Delay {
+    syst: SYST,
+    clock_speed_mhz: u32,
+}
+
+impl Delay {
+    pub fn new(mut syst: SYST, clock_speed_hz: u32) -> Self {
```

Comment:
We should probably add the actual clock generation to the HAL as well, but we can always do this at a later point :wink: